samples/net/sockets/http_server/src/server.c

```diff
@@ -0,0 +1,979 @@
+			LOG_ERR("ERROR writing to socket");
+			return -errno;
+		}
+	}
+
+	unsigned int frame_size = determine_frame_size(ctx_client->buffer);
+
+	ctx_client->offset -= frame_size;
+	memmove(ctx_client->buffer, ctx_client->buffer + frame_size,
+		ctx_client->offset);
+
+	if (ctx_client->offset == 0) {
+		if (ctx_client->streams[ctx_client->stream_count]
+				.stream_state == OPEN_STATE) {
+			handle_http2_open_state(ctx_client);
+		} else {
+			ctx_client->server_state = HTTP2_FRAME_HEADER_STATE;
+		}
+	} else {
+		ctx_client->server_state =
+			determine_server_state(ctx_client->buffer);
+	}
+
+	return 0;
+}
+
+int handle_http2_frame_window_update(struct http2_client_ctx *ctx_client)
+{
+	printf("HTTP2_FRAME_WINDOW_UPDATE\n");
+
+	parse_http2_frame(ctx_client->buffer, ctx_client->offset, &frame);
+	print_http2_frames(&frame);
+
+	unsigned int frame_size = determine_frame_size(ctx_client->buffer);
+
+	ctx_client->offset -= frame_size;
+	memmove(ctx_client->buffer, ctx_client->buffer + frame_size,
+		ctx_client->offset);
+
+	if (!has_upgrade_header) {
+		ctx_client->streams[0].stream_state = IDLE_STATE;
+		ctx_client->stream_count = 1;
+		handle_http2_idle_state(ctx_client);
+		return 0;
+	}
+
+	ctx_client->server_state = HTTP2_FRAME_HEADER_STATE;
+
+	return 0;
+}
+
+int handle_http2_frame_goaway(struct http2_server_ctx *ctx_server,
+	struct http2_client_ctx *ctx_client, int client_index)
+{
+	printf("HTTP2_FRAME_GOAWAY\n");
+
+	parse_http2_frame(ctx_client->buffer, ctx_client->offset, &frame);
+	print_http2_frames(&frame);
+
+	close_client_connection(ctx_server, client_index);
+	has_upgrade_header = true;
+	memset(ctx_client->buffer, 0, sizeof(ctx_client->buffer));
+
+	ctx_client->offset = 0;
+
+	return 0;
+}
+
+int handle_http2_frame_rst_frame(struct http2_server_ctx *ctx_server,
+	struct http2_client_ctx *ctx_client, int client_index)
+{
+	printf("FRAME_RST_STREAM\n");
+
+	parse_http2_frame(ctx_client->buffer, ctx_client->offset, &frame);
+	print_http2_frames(&frame);
+
+	unsigned int frame_size = determine_frame_size(ctx_client->buffer);
+
+	ctx_client->offset -= frame_size;
+	memmove(ctx_client->buffer, ctx_client->buffer + frame_size,
+		ctx_client->offset);
+
+	if (ctx_client->offset == 0) {
+		close_client_connection(ctx_server, client_index);
+		has_upgrade_header = true;
+		memset(ctx_client->buffer, 0, sizeof(ctx_client->buffer));
+
+		ctx_client->server_state = HTTP2_FRAME_HEADER_STATE;
+	} else {
+		ctx_client->server_state =
+			determine_server_state(ctx_client->buffer);
+	}
+
+	return 0;
+}
+
+int on_header_field(struct http_parser *parser, const char *at, size_t length)
+{
+	if (length == 7 && strncasecmp(at, "Upgrade", length) == 0) {
+		LOG_INF("The \"Upgrade: h2c\" header is present.\n");
+		has_upgrade_header = false;
+	}
+	return 0;
+}
+
+int on_url(struct http_parser *parser, const char *at, size_t length)
+{
+	strncpy(url_buffer, at, length);
+	url_buffer[length] = '\0';
+	printf("Requested URL: %s\n", url_buffer);
```

Comment:
Why do we mix printf() and LOG_xxx() macros? Better to use LOG_xxx macros allways to get consistent output.
The printf() could be converted to LOG_DBG()